package-system/AWSNativeSDK/docker_build_aws_sdk.sh

```diff
@@ -6,85 +6,188 @@
+    exit 1
+fi
 
-# Make sure the install path is clear
-inst_path=install
-echo "Command: rm -rf $inst_path"
-rm -rf $inst_path || (echo "Command: rm -rf $inst_path failed" ; exit 1)
-mkdir $inst_path
+BUILD_PATH=$WORKSPACE/aws/build
+if [ -d ${BUILD_PATH} ]
+then
+    rm -rf ${BUILD_PATH}
+fi
 
+INSTALL_PATH=${BUILD_FOLDER}
+if [ -d ${INSTALL_PATH} ]
+then
+    rm -rf ${INSTALL_PATH}
+fi
 
 configure_and_build() {
 
     lib_type=$1
-    build_shared=OFF
+    dep_curl_lib=${CURL_INSTALL}/lib/libcurl.a
     if [ "$lib_type" == "Shared" ]
     then
         build_shared=ON
+    else
+        build_shared=OFF
     fi
 
     echo "CMake Configure $build_type $lib_type"
 
-    cmake -S "$src_path" -B "$bld_path/${lib_type}" \
+    echo cmake -S ${SRC_PATH} -B ${BUILD_PATH}/${lib_type} \
           -G "Ninja" \
           -DTARGET_ARCH=LINUX \
           -DCMAKE_C_COMPILER=/usr/lib/llvm-12/bin/clang \
           -DCMAKE_CXX_COMPILER=/usr/lib/llvm-12/bin/clang++ \
           -DCMAKE_CXX_STANDARD=17 \
           -DCPP_STANDARD=17 \
-          -DCMAKE_C_FLAGS="-fPIC" \
-          -DCMAKE_CXX_FLAGS="-fPIC" \
+          -DCMAKE_C_FLAGS="-fPIC -Wno-option-ignored" \
+          -DCMAKE_CXX_FLAGS="-fPIC  -Wno-option-ignored" \
           -DENABLE_TESTING=OFF \
           -DENABLE_RTTI=ON \
           -DCUSTOM_MEMORY_MANAGEMENT=ON \
           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" \
           -DBUILD_SHARED_LIBS=$build_shared \
           -DCMAKE_BUILD_TYPE=Release \
           -DCMAKE_INSTALL_BINDIR="bin" \
-          -DCMAKE_INSTALL_LIBDIR="lib" 
+          -DCMAKE_INSTALL_LIBDIR="lib" \
+          -DCMAKE_MODULE_PATH=${DOWNLOADED_PACKAGE_FOLDERS} \
+          -DCURL_INCLUDE_DIR=${CURL_INSTALL}/include \
+          -DCURL_LIBRARY=${dep_curl_lib}
+
+    cmake -S ${SRC_PATH} -B ${BUILD_PATH}/${lib_type} \
+          -G "Ninja" \
+          -DTARGET_ARCH=LINUX \
+          -DCMAKE_C_COMPILER=/usr/lib/llvm-12/bin/clang \
+          -DCMAKE_CXX_COMPILER=/usr/lib/llvm-12/bin/clang++ \
+          -DCMAKE_CXX_STANDARD=17 \
+          -DCPP_STANDARD=17 \
+          -DCMAKE_C_FLAGS="-fPIC -Wno-option-ignored" \
+          -DCMAKE_CXX_FLAGS="-fPIC -Wno-option-ignored" \
+          -DENABLE_TESTING=OFF \
+          -DENABLE_RTTI=ON \
+          -DCUSTOM_MEMORY_MANAGEMENT=ON \
+          -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" \
+          -DBUILD_SHARED_LIBS=$build_shared \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_BINDIR="bin" \
+          -DCMAKE_INSTALL_LIBDIR="lib" \
+          -DCMAKE_MODULE_PATH=$DOWNLOADED_PACKAGE_FOLDERS \
+          -DCURL_INCLUDE_DIR=${CURL_INSTALL}/include \
+          -DCURL_LIBRARY=${dep_curl_lib}
```

Comment:
Yeah this is definitely a candidate for use with `set -o xtrace` and `set +o xtrace`